.github/workflows/build_latest.yml

```diff
@@ -63,13 +63,14 @@ jobs:
     - name: Install python dependencies via pip
       run: |
         python -m pip install --upgrade pip
-        pip install numpy cython cftime pytest twine wheel check-manifest mpi4py typing-extensions
+        python -m pip pip install numpy cython cftime pytest twine wheel check-manifest mpi4py typing-extensions
 
     - name: Install netcdf4-python
       run: |
         export PATH=${NETCDF_DIR}/bin:${PATH} 
         export NETCDF_PLUGIN_DIR=${{ github.workspace }}/netcdf-c-${NETCDF_VERSION}/plugins/plugindir
-        python setup.py install
+        python -m pip install .
```

Comment:
@jswhit this is the only part that we need to fix, but I confess I got lost with the parallel errors here. Any advice?

Reply:
I think this is due to the fact that mpi4py is required by setup.py to find the right includes.  Adding mpi4py to 'requires' in pyproject.py fixes it for me - but I don't think that's the right solution.  Is there a way to specify an optional build-system dependency?